sim/firesim-lib/src/main/scala/passes/EC2F1Artefacts.scala

```diff
@@ -31,7 +31,7 @@ object EC2F1Artefacts extends Transform {
   // Capture FPGA-toolflow related verilog defines
   def generateHostVerilogHeader(targetDir: File) {
     val headerName = "cl_firesim_generated_defines.vh"
-    writeOutputFile(new File(targetDir, headerName), s"\n")
+    writeOutputFile(new File(targetDir, headerName), s"`define SYNTHESIS\n")
```

Comment:
@davidbiancolin I'm not sure that we really want to do this here because this is supposed to be things that the host platform requires.  Anyway, just wanted to cal it out

Reply:
This seems like a reasonable place.

Reply:
I was looking at my generated verilog yesterday and noticed that the emitted Verilog for assertions uses `ifundef SYNTHESIS`.  I need to look at it more closely this morning and make sure that `SYNTHESIS` isn't causing the assertions to be synthesized incorrectly in FireSim.

Reply:
As I thought, [Vivado seems to define SYNTHESIS on it's own](https://forums.xilinx.com/t5/Synthesis/Does-vivado-define-word-quot-SYNTHESIS-quot-automatically-How/m-p/815280#M23524)

Most other synthesis tools I've used in the past did this as well, I just didn't find a reference that indicated Vivado did the right thing before I added it.  

@davidbiancolin, I think we should probably back this one line out, what do you think?
```suggestion
    writeOutputFile(new File(targetDir, headerName), s"\n")
```

Reply:
Yup, let's apply this change.